cdap-ui/app/cdap/components/PluginJSONCreator/Create/Content/WidgetCollection/WidgetAttributesCollection/WidgetAttributeInput/index.tsx

```diff
@@ -0,0 +1,156 @@
+/*
+ * Copyright © 2020 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+import WidgetWrapper from 'components/ConfigurationGroup/WidgetWrapper';
+import PluginInput from 'components/PluginJSONCreator/Create/Content/PluginInput';
+import AttributeKeyValueInput from 'components/PluginJSONCreator/Create/Content/WidgetCollection/WidgetAttributesCollection/WidgetAttributeInput/AttributeKeyValueInput';
+import AttributeMultipleValuesInput from 'components/PluginJSONCreator/Create/Content/WidgetCollection/WidgetAttributesCollection/WidgetAttributeInput/AttributeMultipleValuesInput';
+import * as React from 'react';
+
+const WidgetAttributeInput = ({
+  field,
+  fieldInfo,
+  widgetToAttributes,
+  setWidgetToAttributes,
+  widgetID,
+  widgetType,
+}) => {
+  const onAttributeChange = (newVal) => {
+    setWidgetToAttributes((prevObjs) => ({
+      ...prevObjs,
+      [widgetID]: { ...prevObjs[widgetID], [field]: newVal },
+    }));
+  };
+
+  const renderAttributeInput = () => {
+    if (!fieldInfo) {
+      return;
+    }
+    const codeEditors = [
+      'javascript-editor',
+      'json-editor',
+      'python-editor',
+      'scala-editor',
+      'sql-editor',
+    ];
+    if (field === 'default' && codeEditors.includes(widgetType)) {
+      return (
+        <WidgetWrapper
+          widgetProperty={{
+            field,
+            name: field,
+            'widget-type': widgetType,
+            'widget-attributes': {},
+          }}
+          pluginProperty={{
+            required: fieldInfo.required,
+            name: 'code-editor',
+          }}
+          value={widgetToAttributes[widgetID][field]}
+          onChange={onAttributeChange}
+        />
+      );
+    }
+
+    const isMultipleInput = fieldInfo.type.includes('[]');
+    const supportedTypes = fieldInfo.type.split('|');
+    if (!isMultipleInput) {
+      switch (fieldInfo.type) {
+        case 'string':
+          return (
+            <PluginInput
+              widgetType={'textbox'}
```

Comment:
All the cases seem to use the same component and same props. Can we have a helper function that will generate these props based on the field type and remove this switch entirely?

Reply:
I made a helper function named `generateAttributeInput` to generate these props based on the field type.